### PR TITLE
Update hyundai_kia_generic.dbc

### PR DESCRIPTION
**Pull request by xps-genesis**
New DBC from @TK211X merged HKG_WIP_V2_203107122020.dbc into hyundai_kia_generic.dbc

### CODE REVIEW

File: hyundai_kia_generic.dbc [hunk 1]
```diff
@@ -1451,6 +1450,17 @@ BO_ 909 FCA11: 8 FCA
...
+ SG_ Counter : 5|4@0+ (1,0) [0|15] "" XXX
+ SG_ NEW_SIGNAL_1 : 1|2@0+ (1,0) [0|255] "" XXX
+ SG_ NEW_SIGNAL_6 : 16|2@1+ (1,0) [0|3] "" XXX
+ SG_ NEW_SIGNAL_3 : 49|2@0+ (1,0) [0|1] "" XXX
+ SG_ NEW_SIGNAL_7 : 18|14@1+ (1,0) [0|63] "" XXX
+ SG_ NEW_SIGNAL_8 : 15|8@0+ (1,0) [0|255] "" XXX
+ SG_ NEW_SIGNAL_2 : 50|14@1- (1,-4095) [0|16383] "" XXX
+ SG_ NEW_SIGNAL_4 : 34|14@1+ (1,0) [0|16383] "" XXX
+ SG_ NEW_SIGNAL_5 : 33|2@0+ (1,0) [0|1] "" XXX
+ SG_ NEW_SIGNAL_9 : 7|2@0+ (1,0) [0|3] "" XXX
```
Comment (adeebshihadeh) [inline hyundai_kia_generic.dbc:1463]: There's no need to define these signals if we don't know what they are.

File: hyundai_kia_generic.dbc [hunk 2]
```diff
@@ -1466,15 +1476,191 @@ BO_ 905 SCC14: 8 SCC
...
+
+BO_ 1294 LABEL3: 8 XXX
+
+BO_ 1348 544LABEL2: 8 XXX
+
+BO_ 1470 5beLABEL1: 8 XXX
+
+BO_ 1225 4c9LABEL4: 8 XXX
+
+BO_ 549 225: 8 XXX
```
Comment (adeebshihadeh) [inline hyundai_kia_generic.dbc:1666]: Why are these messages defined with no signals?

File: hyundai_kia_generic.dbc (hunk 1 above)
Comment (TK211X) [reply hyundai_kia_generic.dbc:1463]: It allows you to know for sure what the message boundary is. These are actually lane line outputs and the vehicle position relative to them. I have yet to figure out what lane each message follows.

File: hyundai_kia_generic.dbc (hunk 2 above)
Comment (xps-genesis) [reply hyundai_kia_generic.dbc:1666]: These were simply merged from @TK211X's latest. if you don't recommend adding unfinished messages to this PR I can drop them. let me know

File: hyundai_kia_generic.dbc (hunk 1 above)
Comment (adeebshihadeh) [reply hyundai_kia_generic.dbc:1463]: That's fine, just give them a name that indicates what we think they are and maybe add a comment line

File: hyundai_kia_generic.dbc (hunk 2 above)
Comment (adeebshihadeh) [reply hyundai_kia_generic.dbc:1666]: Yes, let's remove all the empty message definitions

File: hyundai_kia_generic.dbc [hunk 3]
```diff
@@ -1466,15 +1476,191 @@ BO_ 905 SCC14: 8 SCC
...
- SG_ LFA_SysWarning : 16|2@1+ (1,0) [0|3] "" XXX
- SG_ ACTIVE2 : 4|2@0+ (1,0) [0|3] "" XXX
  SG_ HDA_USM : 0|2@1+ (1,0) [0|3] "" XXX
+ SG_ ACTIVE2 : 4|2@0+ (1,0) [0|3] "" XXX
+ SG_ LFA_SysWarning : 16|2@1+ (1,0) [0|3] "" XXX
  SG_ ACTIVE : 25|1@1+ (1,0) [0|3] "" XXX
+ SG_ LFA_USM : 28|2@1+ (1,0) [0|3] "" XXX
+
 
+VAL_ 909 CF_VSM_Warn 2 "FCW" 3 "AEB";
```
Comment (adeebshihadeh) [inline hyundai_kia_generic.dbc:1487]: Value lines generally go at the bottom of the DBC

File: hyundai_kia_generic.dbc (hunk 2 above)
Comment (xps-genesis) [reply hyundai_kia_generic.dbc]: Done

File: hyundai_kia_generic.dbc [hunk 4]
```diff
@@ -1466,15 +1477,165 @@ BO_ 905 SCC14: 8 SCC
...
+ SG_ CF_Mdps_Chksum : 63|8@0+ (1,0) [0|255] "" XXX
+ SG_ CR_Mdps_DrvTq : 8|12@1+ (1,0) [0|15] "" XXX
+
+BO_ 357 S_MDPS12: 8 XXX
+ SG_ Counter : 48|4@1+ (1,0) [0|15] "" XXX
+ SG_ Checksum : 63|8@0+ (1,0) [0|255] "" XXX
+ SG_ NEW_SIGNAL_2 : 12|12@1+ (1,0) [0|4095] "" XXX
+ SG_ NEW_SIGNAL_1 : 0|12@1+ (1,0) [0|4095] "" XXX
+
+BO_ 352 AHB1: 8 iBAU
```
Comment (adeebshihadeh) [inline hyundai_kia_generic.dbc:1562]: Can you sort all the new messages by their address?

Comment (xps-genesis) [reply hyundai_kia_generic.dbc:1562]: done! I also updated the alt accel pedal to speed.

File: hyundai_kia_generic.dbc [hunk 5]
```diff
@@ -1466,15 +1477,165 @@ BO_ 905 SCC14: 8 SCC
...
+
+BO_ 865 ADAS_PRK_11: 8 ADAS_PRK
+ SG_ CF_PCA_BrkReq : 24|1@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ CF_PCA_DclTrgtVal : 28|4@1+ (0.04,0) [0|0] "g" Vector__XXX
+ SG_ PCA_ALIVE_CNT : 40|4@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ PCA_CHECK_SUM : 48|8@1+ (1,0) [0|0] "" Vector__XXX
+
+BO_ 882 EV_PC5: 8 CGW
+ SG_ VS : 0|8@1+ (1,0) [0|255] "" XXX
+ SG_ CF_Vcu_GearSelDisp : 16|4@1+ (1,0) [0|15] "" Vector__XXX
```
Comment (adeebshihadeh) [inline hyundai_kia_generic.dbc:1514]: What's the extra bit for? The previous signal was 3 bits.

Comment (xps-genesis) [reply hyundai_kia_generic.dbc:1514]: looks like D is 5 and S is 8. so 8 needs 4 bits(1 0 0 0)

Comment (TK211X) [reply hyundai_kia_generic.dbc:1514]: If you ever see the extra bit active and you make a wish, it will come true.

File: hyundai_kia_generic.dbc [hunk 6]
```diff
@@ -1466,15 +1477,165 @@ BO_ 905 SCC14: 8 SCC
...
+ SG_ CR_Mcu_MotEstTq_Pc : 33|12@1- (0.048828125,0) [-100|99.951171875] "%" Vector__XXX
+
+BO_ 881 EV_PC4: 8 CGW
+ SG_ CR_Brk_StkDep_Pc : 0|8@1+ (1,0) [0|0] "%" Vector__XXX
+ SG_ CF_Vcu_Rdy : 9|1@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ CF_Vcu_EvRdyClu : 10|1@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ CF_Vcu_EvRdy : 11|1@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ CF_Vcu_EcoMdOff : 12|1@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ Unknown_Paint1 : 18|13@1+ (1,0) [0|63] "" XXX
+ SG_ CR_Vcu_VehSpeed : 31|8@1+ (0.55,0) [0|100] "kph or mph" Vector__XXX
```
Comment (adeebshihadeh) [inline hyundai_kia_generic.dbc:1525]: I plotted this in cabana, and it's clearly not vehicle speed.

Comment (xps-genesis) [reply hyundai_kia_generic.dbc:1525]: which vehicle route did you use? I have tested on Ioniq and Kona EV. both showed correct values

https://my.comma.ai/cabana/?route=5399c75340359d11%7C2020-07-19--03-21-10&exp=1626666945&sig=5PoY1lEVkfVangCnmH0UATmYXTSXO4vC66gSSGrJrcw%3D&max=21&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F5399c75340359d11%2F0affdbb847264ac22750f32532087c4f_2020-07-19--03-21-10

Comment (xps-genesis) [reply hyundai_kia_generic.dbc:1525]: image

**Comment by xps-genesis** [comment]
KONA EV matches current DBC. changes to EV messages not required!
https://my.comma.ai/cabana/?route=6d556f411feddcac%7C2020-07-27--14-18-15&exp=1627524848&sig=awMmAKOJ91Vw7HSBjliICjA9r1fN85lLgHkhz18T5O4%3D&max=67&url=https%3A%2F%2Fchffrprivate-vzn.azureedge.net%2Fchffrprivate3%2Fv2%2F6d556f411feddcac%2Ffd5190e1aaea0c9a47ffbf00c5b1b8f2_2020-07-27--14-18-15

File: hyundai_kia_generic.dbc (hunk 6 above)
Comment (adeebshihadeh) [reply hyundai_kia_generic.dbc]: Current EV gas signal is correct

**Pull request merged**